scalafix-cli/src/main/scala/scalafix/internal/jgit/JGitDiff.scala

```diff
@@ -0,0 +1,103 @@
+package scalafix.internal.jgit
+
+import java.nio.file.Path
+
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.api.errors.GitAPIException
+import org.eclipse.jgit.diff.DiffEntry.ChangeType._
+import org.eclipse.jgit.diff.DiffFormatter
+import org.eclipse.jgit.patch.FileHeader
+import org.eclipse.jgit.lib.ObjectReader
+import org.eclipse.jgit.lib.Ref
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.ProgressMonitor
+import org.eclipse.jgit.dircache.DirCacheIterator
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevTree
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.treewalk.AbstractTreeIterator
+import org.eclipse.jgit.treewalk.CanonicalTreeParser
+import org.eclipse.jgit.treewalk.FileTreeIterator
+import org.eclipse.jgit.util.io.NullOutputStream
+import scala.collection.JavaConverters._
+
+object JGitDiff {
+  def apply(workingDir: Path, baseBranch: String): List[GitDiff] = {
+    val builder = new FileRepositoryBuilder()
+    val repository =
+      builder.readEnvironment().setWorkTree(workingDir.toFile).build()
+
+    val baseBranchRef = s"refs/heads/$baseBranch"
+    val git = new Git(repository)
+    val branches = git.branchList().call()
+    if (branches.isEmpty) {
+      throw new Exception(s"$workingDir is not a git repository")
```

Comment:
If we use exceptions we should try to create custom exception types instead of java.lang.Exception so that these can be caught without string fiddling.